x/emissions/module/rewards/scores.go

```diff
@@ -108,33 +109,25 @@ func GenerateReputerScores(
 		if err != nil {
 			return []types.Score{}, errors.Wrapf(err, "Error inserting reputer score")
 		}
-		previousScore, err := keeper.GetReputerScoreEma(ctx, topicId, reputer)
-		if err != nil {
-			return []types.Score{}, errors.Wrapf(err, "Error getting reputer score ema")
-		}
-		firstTime := previousScore.BlockHeight == 0 && previousScore.Score.IsZero()
-		emaScoreDec, err := alloraMath.CalcEma(
-			topic.MeritSortitionAlpha,
-			newScore.Score,
-			previousScore.Score,
-			firstTime,
-		)
-		if err != nil {
-			return []types.Score{}, errors.Wrapf(err, "Error calculating ema")
-		}
-		emaScore := types.Score{
-			TopicId:     topicId,
-			BlockHeight: block,
-			Address:     reputer,
-			Score:       emaScoreDec,
-		}
-		err = keeper.SetReputerScoreEma(ctx, topicId, reputer, emaScore)
+
+		err = keeper.CalcAndSaveReputerScoreEmaIfNewUpdate(ctx, topic, block, reputer, newScore)
```

Comment:
Here, it’s calculating and saving the scores' EMA, and right above, it’s inserting the "normal" scores into the `reputerScoresByBlock` KV store. Apparently, this KV store it’s not being used by anything apart from the query server (which I think should use the EMA kv store instead). Need to double-check if this  it’s still needed.

Reply:
Also the quantile is not being calculated using the EMA score, but the "normal"score

Reply:
fixed the quantile calculation